### ISSUE TITLE
Second assignment of array variable in user_nl doesn't overwrite the first setting

### ISSUE THREAD

**Issue by billsacks** [ESMCI/cime#1397]
I set up a case with

```
./create_newcase -case test_nml_0420a -compset A -res f45_g37 --run-unsupported
```

And put the following in `user_nl_cpl`:

```
cplflds_custom = 'foo','bar'
cplflds_custom = 'baz'
```

After running `./case.setup` and `./preview_namelists`, I expected that the final value of `cplflds_custom` in `CaseDocs/drv_in` would be `baz`, but in fact I found:

```
  cplflds_custom = 'baz', 'bar'
```

In practice, this is unlikely to affect people very often, but we do sometimes rely on this double-setting for some recursive testmods.

**Comment by erichlf** [comment]
I have been looking at the code to see how to address this issue, however it appears that this really is intended and I am not sure there is an easy fix for it due to the following intended logic 
`>>> merge_literal_lists(['3*false', '3*true'], ['true', '4*', 'false'])          
      ['true', 'false', 'false', 'true', 'true', 'false']`

**Comment by jedwards4b** [comment]
@billsacks - I think that this code is working as intended.

**Comment by billsacks** [comment]
I'm not familiar with the internals of the implementation, so would rather talk about the intended behavior from a user perspective.

@jedwards4b , to be clear, you're saying that the intended behavior is that, if you put the following in user_nl_cpl:

```
cplflds_custom = 'foo','bar'
cplflds_custom = 'baz'
```

then you'll end up with the following?:

```
cplflds_custom = 'baz', 'bar'
```

I find that unintuitive: If I specify `cplflds_custom = 'baz'`, then I expect to see just 'baz'. But I'm willing to let it go if there is truly a use case that requires that behavior.

@erichlf : If the current behavior is truly the intended behavior, does that mean that you need to back out a change you made in #1389 ?

**Comment by erichlf** [comment]
I would back out a change that didn't work outside the unit test.

**Comment by jedwards4b** [comment]
Yes this is the intended behavior - cplflds_custom is an array with some default values, 
say
cplflds_custom = 'one', two', 'three'
the first occurrence of cplflds_custom in the user_nl_cpl will update that to be
cplflds_custom = 'foo', 'bar', 'three'
and the second
cplflds_custom = 'baz', 'bar', 'three'

I think that we need to maintain the current behavior for overriding the defaults and what you are asking for is that if a variable occurs in the user_nl_* file more than once then only the final occurrence of that variable should be processed at all.   Is that correct?  Can you think of any exceptions?

**Comment by billsacks** [comment]
@jedwards4b - thanks for clarifying this use case. It does leave me wondering, though: is this really something that people want to do in practice? I would imagine that, in practice, it would be much more common to either want to:

(1) Add new fields to cplflds_custom - as will be possible with #1389 , using `+=`

(2) Overwrite a value at a particular index; e.g.:

```
cplflds_custom(1) = 'foo'
```

The example you gave would be a use case where the user wants to overwrite the first N values of cplflds_custom, leaving the existing values (N+1...) as is. I can't imagine many scenarios where I'd want to do that, with this variable or other list variables: it seems unlikely that the values I want to overwrite all occur at the start of the list.

That said: If you or others feel this is an important use case to maintain, I'm really fine with it. I guess it's right that what I'm then asking in this issue is:

> if a variable occurs in the user_nl_* file more than once then only the final occurrence of that variable should be processed at all

However, I don't feel this is critical, so I'm okay closing this as a wontfix: I opened this issue thinking it was a bug, but since it is apparently the intended behavior I don't feel strongly about it.

**Comment by gold2718** [comment]
I think that the behavior in a user_nl_* file should match what happens in a Fortran namelist file.
@jedwards4b's explanation is correct. I would rephrase to say:
Each line in a namelist is processed in order.
For setting arrays, each line sets some (or all) elements of the array and leaves others untouched.

**Comment by billsacks** [comment]
It seems this is working as intended, so I'm closing it as invalid.
